core/src/main/codegen/templates/Parser.jj

```diff
@@ -4270,21 +4291,14 @@ SqlIdentifier CompoundIdentifier() :
 {
     List<String> list = new ArrayList<String>();
     List<SqlParserPos> posList = new ArrayList<SqlParserPos>();
-    String p;
+    List<Boolean> quotes = new ArrayList<Boolean>();
     boolean star = false;
 }
 {
-    p = Identifier()
-    {
-        posList.add(getPos());
-        list.add(p);
-    }
+    IdentifierSegment(list, posList, quotes)
     (
         <DOT>
-        p = Identifier() {
-            list.add(p);
-            posList.add(getPos());
-        }
+        IdentifierSegment(list, posList, quotes)
```

Comment:
I wonder if we should be consistent after `<DOT>`: using`Identifier` (or after the PR `IdentifierSegment`) vs. using `SimpleInditifier`. Since we now have a unified code path for doing so.

I found that some of the places are giving inconsistent results. For example:
* `tbl.foo(0).col.bar` is parsed as: ``` `TBL`.`COL`.`BAR` ```
* `tbl.foo(0).col.bar` is parsed as: ``` ((`TBL`.`FOO`(0).`COL`).`BAR`) ```

Of course this can also be addressed in a separated PR.

Reply:
@walterddr Well, i think
> * `tbl.foo(0).col.bar` is parsed as: `` ((`TBL`.`FOO`(0).`COL`).`BAR`) ``

I think for after this PR, the right way to quote this id is tbl.`foo(0)`.col.bar

Reply:
yes. the correct way should be ``` `TBL`.`FOO`(0).`COL`.`BAR` ```. However it was incorrectly parsed before this PR already, so I don't think it should block this one :-)